src/pickers/HealthFacilityReferPicker.js

```diff
@@ -0,0 +1,73 @@
+import { healthFacilityLabel, LOCATION_SUMMARY_PROJECTION } from "../utils";
+import React, { useState } from "react";
+import { useSelector } from "react-redux";
+import { useModulesManager, useTranslations, Autocomplete, useGraphqlQuery } from "@openimis/fe-core";
+import _debounce from "lodash/debounce";
+
+const HealthFacilityReferPicker = (props) => {
+  const {
+    onChange,
+    readOnly,
+    required,
+    withLabel = true,
+    withPlaceholder,
+    value,
+    label,
+    filterOptions,
+    filterSelectedOptions,
+    placeholder,
+    multiple,
+    region,
+    district,
+    level,
+  } = props;
+
+  const modulesManager = useModulesManager();
+  const { formatMessage } = useTranslations("location", modulesManager);
+  const [searchString, setSearchString] = useState("");
+  const { data, isLoading, error } = useGraphqlQuery(
+    `
+    query HealthFacilityPicker ($str: String, $region: String, $district: [String], $level: String) {
+      healthFacilities: healthFacilitiesStr(first: 20, str: $str, regionUuid: $region, districtsUuids: $district, level: $level) {
+        edges {
+          node {
+            id
+            uuid
+            code
+            name
+            level
+            servicesPricelist{id, uuid}
+            itemsPricelist{id, uuid}
+            location {${LOCATION_SUMMARY_PROJECTION.join(",")} parent { ${LOCATION_SUMMARY_PROJECTION.join(",")} }}
```

Comment:
This should be also adjusted to your use-case. Since you're not using locations here, consider removing them. (I didn't check if location is obligatory, so maybe first of all check it out.)